src/main/java/cms/gov/madie/measure/services/ActionLogService.java

```diff
@@ -85,4 +86,25 @@ public boolean logMeasureSetAction(
             .build(),
         collection);
   }
+
+  public List<Action> findMeasureHistory(String measureId, String measureSetId) {
+    ActionLog measureActionLogs = measureActionLogRepository.findByTargetId(measureId);
+    Optional<MeasureSetActionLog> measureSetActionLogs =
+        measureSetActionLogRepository.findByTargetId(measureSetId);
+
+    List<Action> combinedActionLogs = new ArrayList<>();
+
+    if (measureActionLogs != null && !CollectionUtils.isEmpty(measureActionLogs.getActions())) {
+      combinedActionLogs.addAll(measureActionLogs.getActions());
+    }
+
+    measureSetActionLogs.ifPresent(
+        log -> {
+          if (log.getActions() != null && !log.getActions().isEmpty()) {
```

Comment:
```suggestion
          if (CollectionUtils.isNotEmpty(log.getActions())) {
```
Recommend simplifying with CollectionUtils.

Reply:
done